### DISCUSSION TITLE
DCE Demo Discussions: `node-projects-dce` by Jochen Kühner

### DISCUSSION THREAD

**Discussion by Westbrook** [w3c/webcomponents-cg#91] [Ideas]
As presented on 25 March to the WCCG, `node-projects-dce` is a proposed approach to declarative custom elements by @jogibear9988.

Links TK.

Further discussion and questions below.

**Comment by jogibear9988** [comment]
Sample is here: [Sample](https://node-projects.github.io/web-component-designer-demo/index.html?html=%3Cnode-projects-dce%20name=%22simple-dce-demo%22%20properties=%27{%22list%22:%22Array%22,%20%22list2%22:%22Array%22,%20%22ctx%22:{%22type%22:%22String%22,%22reflect%22:true}}%27%20enable-bindings%20%3E%20%3Ctemplate%3E%20%3Cstyle%3Eh1%20{color:%20red}%3C/style%3E%20%3Ch1%3EHello%20World%3C/h1%3E%20%3Cdiv%20style=%22border:%20solid%203px%20black%22%3ECtx:%20[[this.ctx]]%3C/div%3E%20%3Ctemplate%20repeat:myitem=%22[[this.list]]%22%3E%20%3Cbutton%3E[[myitem.toUpperCase()]]%20-%20%3Cb%3E[[myitem.toLowerCase()]]%3C/b%3E%20-%20[[index]]%3C/button%3E%20%3Cul%3E%20%3Ctemplate%20repeat:myitem2=%22[[this.list2]]%22%20repeat-index=%22inneridx%22%3E%20%3Cbutton%20@click=%22[[this.ctx%20=%20myitem2]]%22%20%3E[[myitem.toUpperCase()]]%20-%20%3Cb%3E[[myitem2.toLowerCase()]]%3C/b%3E%20-%20[[inneridx%20*%20100]]%3C/button%3E%20%3C/template%3E%20%3C/ul%3E%20%3C/template%3E%20%3C/template%3E%20%3C/node-projects-dce%3E%20%3Csimple-dce-demo%20list=%27[%22aa%22,%22bb%22,%22cc%22]%27%20list2=%27[%22hello%22,%20%22you%22]%27%20ctx=%22TestCtx%22%20style=%22position:absolute;left:184px;top:-53px;%22%3E%3C/simple-dce-demo%3E)
